src/irreps/dnirrep.jl

```diff
@@ -0,0 +1,279 @@
+        # A_i x A_j = A_{xor(i,j)}
+        if iszero(b.j) # then also iszero(a.j)
+            cj = 0
+            cisodd = xor(a.isodd, b.isodd)
+            return DNIrrep{N}(cj, cisodd), 5
+        end
+
+        if iszero(a.j)
+            cj = b.j
+
+            # A_i x B_j = B_{xor(i,j)}
+            if iseven(N) && (cj == N >> 1)
+                cisodd = xor(a.isodd, b.isodd)
+                return DNIrrep{N}(cj, cisodd), 5
+            end
+
+            # A_i x rho_j = rho_j
+            return b, 5
+        end
+
+        if a.j == b.j # != 0
+            # B_i x B_j = A_{xor(i,j)}
+            if iseven(N) && b.j == N >> 1
+                return DNIrrep{N}(0, xor(a.isodd, b.isodd)), 5
+            end
+
+            # rho_i x rho_i = A_1 + A_2 + rho_2i
+            return DNIrrep{N}(0, false), 2
+        end
+
+        # rho_i x B_j = rho_{j-i}
+        if iseven(N) && b.j == (N >> 1)
+            return DNIrrep{N}(b.j - a.j, false), 5
+        end
+
+        # rho_i x rho_j = rho_{i-j} + rho_{i+j}
+        return DNIrrep{N}(b.j - a.j, false), 3
+    elseif state == 2
+        return DNIrrep{N}(0, true), 3
+    elseif state == 3
+        cj = a.j + b.j
+        if iseven(N) && cj == (N >> 1)
+            return DNIrrep{N}(cj, false), 4
+        end
+        if cj > (N >> 1)
+            cj = N - cj
+        end
+        return DNIrrep{N}(cj, false), 5
+    elseif state == 4
+        return DNIrrep{N}(N >> 1, true), 5
+    else
+        return nothing
+    end
+end
+
+# Topological data
+# ----------------
+dim(a::DNIrrep{N}) where {N} = ifelse((a.j == 0) | (iseven(N) & (a.j == (N >> 1))), 1, 2)
+
+function Nsymbol(a::DNIrrep{N}, b::DNIrrep{N}, c::DNIrrep{N}) where {N}
+    j_satisfied = (c.j == min(a.j + b.j, N - (a.j + b.j))) | (c.j == max(a.j, b.j) - min(a.j, b.j))
+    s_satisfied = (dim(c) != 1) | (dim(a) != 1) | (xor(a.isodd, b.isodd) == c.isodd)
+    return j_satisfied & s_satisfied
+end
+
+function Fsymbol(a::I, b::I, c::I, d::I, e::I, f::I) where {N, I <: DNIrrep{N}}
+    T = sectorscalartype(I)
+    (Nsymbol(a, b, e) & Nsymbol(e, c, d) & Nsymbol(b, c, f) & Nsymbol(a, f, d)) || return zero(T)
+
+    # tensoring with units gives 1
+    (isone(a) || isone(b) || isone(c)) && return one(T)
+
+    # fallback through fusiontensor
+    A = fusiontensor(a, b, e)
+    A = reshape(A, size(A, 1), size(A, 2), size(A, 3))
+    B = fusiontensor(e, c, d)
+    B1 = reshape(view(B, :, :, 1), size(B, 1), size(B, 2))
+    C = fusiontensor(b, c, f)
+    C = reshape(C, size(C, 1), size(C, 2), size(C, 3))
+    D = fusiontensor(a, f, d)
+    D1 = reshape(view(D, :, :, 1), size(D, 1), size(D, 2))
+
+    return @tensor conj(D1[1 5]) * conj(C[2 4 5]) * A[1 2 3] * B1[3 4]
+end
+
+function Rsymbol(a::I, b::I, c::I) where {N, I <: DNIrrep{N}}
+    R = convert(sectorscalartype(I), Nsymbol(a, b, c))
+    return ifelse((c.j == 0) & c.isodd & !(a.j == b.j == 0) & !(iseven(N) && (a.j == b.j == (N >> 1))), -R, R)
+end
+
+const _invsqrt2 = 1 / sqrt(2)
```

Comment:
I would move ethat one inside `fusiontensor` and make sure it is of type `sectorscalartype(I)`.